runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+	"OperationBitwiseRightShift": OperationBitwiseRightShift,
+}
+
+var statementKindMap = map[string]func() Statement{
+	"AssignmentStatement":        func() Statement { return &AssignmentStatement{} },
+	"BreakStatement":             func() Statement { return &BreakStatement{} },
+	"CompositeDeclaration":       func() Statement { return &CompositeDeclaration{} },
+	"ContinueStatement":          func() Statement { return &ContinueStatement{} },
+	"EmitStatement":              func() Statement { return &EmitStatement{} },
+	"ExpressionStatement":        func() Statement { return &ExpressionStatement{} },
+	"ForStatement":               func() Statement { return &ForStatement{} },
+	"FunctionDeclaration":        func() Statement { return &FunctionDeclaration{} },
+	"IfStatement":                func() Statement { return &IfStatement{} },
+	"ImportDeclaration":          func() Statement { return &ImportDeclaration{} },
+	"InterfaceDeclaration":       func() Statement { return &InterfaceDeclaration{} },
+	"PragmaDeclaration":          func() Statement { return &PragmaDeclaration{} },
+	"ReturnStatement":            func() Statement { return &ReturnStatement{} },
+	"SpecialFunctionDeclaration": func() Statement { return &SpecialFunctionDeclaration{} },
+	"SwapStatement":              func() Statement { return &SwapStatement{} },
+	"SwitchStatement":            func() Statement { return &SwitchStatement{} },
+	"TransactionDeclaration":     func() Statement { return &TransactionDeclaration{} },
+	"VariableDeclaration":        func() Statement { return &VariableDeclaration{} },
+	"WhileStatement":             func() Statement { return &WhileStatement{} },
+}
+
+var transferOperationMap = map[string]TransferOperation{
+	"TransferOperationUnknown":    TransferOperationUnknown,
+	"TransferOperationCopy":       TransferOperationCopy,
+	"TransferOperationMove":       TransferOperationMove,
+	"TransferOperationMoveForced": TransferOperationMoveForced,
+}
+
+var typeNameMap = map[string]func() Type{
+	"ConstantSizedType": func() Type { return &ConstantSizedType{} },
+	"DictionaryType":    func() Type { return &DictionaryType{} },
+	"FunctionType":      func() Type { return &FunctionType{} },
+	"InstantiationType": func() Type { return &InstantiationType{} },
+	"NominalType":       func() Type { return &NominalType{} },
+	"OptionalType":      func() Type { return &OptionalType{} },
+	"ReferenceType":     func() Type { return &ReferenceType{} },
+	"RestrictedType":    func() Type { return &RestrictedType{} },
+	"VariableSizedType": func() Type { return &VariableSizedType{} },
+}
+
+var variableKindMap = map[string]VariableKind{
+	"VariableKindNotSpecified": VariableKindNotSpecified,
+	"VariableKindVariable":     VariableKindVariable,
+	"VariableKindConstant":     VariableKindConstant,
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+type BigIntUnmarshaler struct{ Int **big.Int }
```

Comment:
Just like with the maps above: Consider moving these unmarshaling types and functions next to their corresponding type / next to the type's JSON marshaling function. 

That way we won't forget to update the unmarshaling code when we update the type (i.e. keep type, marshaling, and unmarshaling code all in sync).

Also, consider putting the fields on separate lines to make the code more readable.